apps/bfd-server-ng/src/main/java/gov/cms/bfd/server/ng/eob/EobHandler.java

```diff
@@ -81,8 +120,94 @@ public Bundle searchById(
 
   private Optional<ExplanationOfBenefit> searchByIdInner(
       Long claimUniqueId, DateTimeRange serviceDate, DateTimeRange lastUpdated) {
-    var claim = claimRepository.findById(claimUniqueId, serviceDate, lastUpdated);
+    var eobs = claimRepository.findById(claimUniqueId, serviceDate, lastUpdated).stream().toList();
+    Bundle bundle = getFhirEobs(eobs, true);
+    return bundle.getEntry().stream()
+        .map(entry -> entry.getResource())
+        .filter(ExplanationOfBenefit.class::isInstance)
+        .map(ExplanationOfBenefit.class::cast)
+        .findFirst();
+  }
+
+  private String getDiagnosisCode(ClaimProcedure e) {
+    return normalize(e.getDiagnosisCode());
+  }
+
+  private String getClaimProcedureCode(ClaimProcedure e) {
+    return normalize(e.getProcedureCode());
+  }
+
+  private String getDictCode(Map<String, Object> entry) {
+    return normalize(Optional.ofNullable(entry.get("code").toString()));
+  }
+
+  private String normalize(Optional<String> code) {
+    return code.orElse("").trim().replace(".", "").toLowerCase();
+  }
+
+  // Returns true if the given claim contains any procedure that matches a SAMHSA
+  // security label code from the provided dictionary and keys.
+  private boolean claimHasSamhsa(Claim claim, Map<String, List<Map<String, Object>>> dict) {
+
+    return drgCodeMatches(claim, dict)
+        || claim.getClaimItems().stream()
+            .anyMatch(
+                e -> {
+                  var cl = e.getClaimLine();
+                  var cp = e.getClaimProcedure();
+
+                  return procedureMatchesDict(cp, dict) || hcpcsCodeMatches(cl, dict);
+                });
+  }
+
+  // Checks DRG.
+  private boolean drgCodeMatches(Claim claim, Map<String, List<Map<String, Object>>> dict) {
+    var entries = dict.getOrDefault(SystemUrls.CMS_MS_DRG, List.of());
```

Comment:
We can use `get` instead of `getOrDefault` since these keys will always be present in the list. If not, we have bigger problems.

```suggestion
    var entries = dict.get(SystemUrls.CMS_MS_DRG);
```